x/staking/keeper/validator.go

```diff
@@ -458,13 +456,30 @@ func (k Keeper) UnbondAllMatureValidators(ctx sdk.Context) {
 					panic("unexpected validator in unbonding queue; status was not unbonding")
 				}
 
-				val = k.UnbondingToUnbonded(ctx, val)
-				if val.GetDelegatorShares().IsZero() {
-					k.RemoveValidator(ctx, val.GetOperator())
+				if val.UnbondingOnHoldRefCount == 0 {
+					for _, id := range val.UnbondingIds {
+						k.DeleteUnbondingIndex(ctx, id)
+					}
+					val = k.UnbondingToUnbonded(ctx, val)
+					if val.GetDelegatorShares().IsZero() {
+						k.RemoveValidator(ctx, val.GetOperator())
+					} else {
+						// remove unbonding ids
+						val.UnbondingIds = []uint64{}
+					}
+					// remove validator from queue
+					k.DeleteValidatorQueue(ctx, val)
```

Comment:
What happened to the `store.Delete(key)` call?

Reply:
@mpoke

Reply:
Instead of dealing the entire entry, we just delete the validator that reaches maturity. See https://github.com/cosmos/cosmos-sdk/blob/caa6904e185d7de37c8605b59cfc37c185b4b1ba/x/staking/keeper/validator.go#L380

Reply:
This enables different validators to reach maturity at different times.

Reply:
But where are we deleting the key? Did we alter `DeleteValidatorQueue` to now delete it?